packages/kbn-expandable-flyout/src/actions.ts

```diff
@@ -20,39 +21,17 @@ export enum ActionType {
   closeFlyout = 'close_flyout',
 }
 
-export type Action =
-  | {
-      type: ActionType.openFlyout;
-      payload: {
-        right?: FlyoutPanelProps;
-        left?: FlyoutPanelProps;
-        preview?: FlyoutPanelProps;
-      };
-    }
-  | {
-      type: ActionType.openRightPanel;
-      payload: FlyoutPanelProps;
-    }
-  | {
-      type: ActionType.openLeftPanel;
-      payload: FlyoutPanelProps;
-    }
-  | {
-      type: ActionType.openPreviewPanel;
-      payload: FlyoutPanelProps;
-    }
-  | {
-      type: ActionType.closeRightPanel;
-    }
-  | {
-      type: ActionType.closeLeftPanel;
-    }
-  | {
-      type: ActionType.closePreviewPanel;
-    }
-  | {
-      type: ActionType.previousPreviewPanel;
-    }
-  | {
-      type: ActionType.closeFlyout;
-    };
+export const openPanelsAction = createAction<{
+  right?: FlyoutPanelProps;
+  left?: FlyoutPanelProps;
+  preview?: FlyoutPanelProps;
+}>(ActionType.openFlyout);
+
+export const openRightPanelAction = createAction<FlyoutPanelProps>(ActionType.openRightPanel);
+export const openLeftPanelAction = createAction<FlyoutPanelProps>(ActionType.openLeftPanel);
+export const openPreviewPanelAction = createAction<FlyoutPanelProps>(ActionType.openPreviewPanel);
+export const closePanelsAction = createAction(ActionType.closeFlyout);
+export const closeRightPanelAction = createAction(ActionType.closeRightPanel);
+export const closeLeftPanelAction = createAction(ActionType.closeLeftPanel);
+export const closePreviewPanelAction = createAction(ActionType.closePreviewPanel);
+export const previousPreviewPanelAction = createAction(ActionType.previousPreviewPanel);
```

Comment:
💅 Consider adding some empty lines in between to let the code breath ;)

Reply:
done!